Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -4491,12 +4493,43 @@ public String doCharacterAttemptAttack(ODPAuthenticator auth, CachedEntity user,
         Double charDex = getCharacterDexterity(sourceCharacter);
         Double monsterDex = getCharacterDexterity(targetCharacter);
         Random rnd = new Random();
+        
+        String status = "";
+        
+        int autoRunThreshold = targetCharacter.getProperty("autoRunThreshold");
+        if(autoRunThreshold == null) autoRunThreshold = 0;
+        int autoRunChance = targetCharacter.getProperty("autoRunChance");
+        if(autoRunChance == null) autoRunChance = 0;
+        
+        //First, we check against the auto run threshold to see if the monster will attempt to run.
+        if((targetCharacter.getProperty("hitpoints") / targetCharacter.getProperty("maxHitpoints"))*100 < autoRunThreshold) {
```

Comment:
getProperty("hitpoints") is not cast to a number so you would have gotten compiler errors here. You can't submit a PR if you have compiler errors so I'll stop looking at this until you bring it up and make sure compiler errors are taken care of first.